File: Templates/Java/src/main/java/Person/Person.java
package Person;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

public class Person {
    @JsonProperty("first_name")
    private String firstName;

    @JsonProperty("last_name")
    private String lastName;

    @JsonProperty("age")
    private int age;

    @JsonProperty("address")
    private Address address;

    @JsonProperty("jobs")
    private List<Job> jobs;

    // Getters and Setters
    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    public List<Job> getJobs() {
        return jobs;
    }

    public void setJobs(List<Job> jobs) {
        this.jobs = jobs;
    }
}

File: Templates/TypeScript/Person/person.ts
import { JsonProperty } from "json-object-mapper";
import { Address } from "./address";
import { Job } from "./job";

export class Person {
	@JsonProperty({ name: "first_name", required: true })
	public firstName: string;

	@JsonProperty({ name: "last_name", required: true })
	public lastName: string;

	@JsonProperty({ required: true })
	public age: number;

	@JsonProperty({ required: true })
	public address: Address;

	@JsonProperty({ required: true })
	public jobs: Job[];

	@JsonProperty({ required: false })
	public parent?: Person;

	@JsonProperty({ required: false })
	public child?: Person;
}
